src/background/telemetry/exception-telemetry-listener.ts

```diff
@@ -0,0 +1,183 @@
+                if (message || err) {
+                    sendExceptionTelemetry(ErrorType.ConsoleError, message, err?.stack);
+                }
+
+                if (optionalParams.length > 1) {
+                    consoleError(message, ...optionalParams);
+                } else {
+                    consoleError(message, err);
+                }
+            } finally {
+                consoleHookIsActive = false;
+            }
+        };
+
+        // Catch errors written to logger.error
+        // Note that this is separate from console.error (despite the default logger deferring to
+        // console.error) because we generated the default logger before updating console.error
+        // above, meaning logger.error still has the old console.error implementation.
+        const loggerError = logger.error;
+        logger.error = function (message?: any, ...optionalParams: any[]) {
+            if (loggingHookIsActive) {
+                return;
+            }
+            loggingHookIsActive = true;
+            try {
+                const err = optionalParams[0] as Error;
+                if (message || err) {
+                    sendExceptionTelemetry(ErrorType.LoggerError, message, err?.stack);
+                }
+
+                if (optionalParams.length > 1) {
+                    loggerError(message, ...optionalParams);
+                } else {
+                    loggerError(message, err);
+                }
+            } finally {
+                loggingHookIsActive = false;
+            }
+        };
+    }
+
+    private sendExceptionTelemetry = (
+        errorType: string,
+        message: string,
+        stackTrace?: string,
+        source?: string,
+    ): void => {
+        const telemetry: UnhandledExceptionTelemetryData = {
+            message: message.toString(),
+            stackTrace,
+            source,
+            errorType,
+        };
+        const sanitizedTelemetry = this.sanitizeTelemetryData(telemetry);
+
+        if (sanitizedTelemetry) {
+            const payload: BaseActionPayload = {
+                telemetry: sanitizedTelemetry,
+            };
+
+            this.telemetryEventHandler.publishTelemetry(
+                TelemetryEvents.UNHANDLED_EXCEPTION,
+                payload,
+            );
+        }
+    };
+
+    private sanitizeTelemetryData = (
+        telemetryData: UnhandledExceptionTelemetryData,
+    ): UnhandledExceptionTelemetryData => {
+        if (telemetryData.message && telemetryData.message.length > this.MAX_MESSAGE_CHARS) {
+            telemetryData.message = telemetryData.message.substring(0, this.MAX_MESSAGE_CHARS);
+        }
+        if (telemetryData.stackTrace && telemetryData.stackTrace.length > this.MAX_STACK_CHARS) {
+            telemetryData.stackTrace = telemetryData.stackTrace.substring(0, this.MAX_STACK_CHARS);
+        }
+        if (
+            (telemetryData.message && telemetryData.message.includes('http')) ||
```

Comment:
I think it would be preferable for this to look for `://` instead of `http`; it's possible for URLs to use different protocols (particularly, `file://`)